src/analysis/convex/normed.lean

```diff
@@ -0,0 +1,143 @@
+(convex_on_dist y (convex_convex_hull ℝ _)).exists_ge_of_mem_convex_hull hx
+
+/-- Given a point `x` in the convex hull of `s` and a point `y` in the convex hull of `t`,
+there exist points `x' ∈ s` and `y' ∈ t` at distance at least `dist x y`. -/
+lemma convex_hull_exists_dist_ge2 {s t : set E} {x y : E}
+  (hx : x ∈ convex_hull ℝ s) (hy : y ∈ convex_hull ℝ t) :
+  ∃ (x' ∈ s) (y' ∈ t), dist x y ≤ dist x' y' :=
+begin
+  rcases convex_hull_exists_dist_ge hx y with ⟨x', hx', Hx'⟩,
+  rcases convex_hull_exists_dist_ge hy x' with ⟨y', hy', Hy'⟩,
+  use [x', hx', y', hy'],
+  exact le_trans Hx' (dist_comm y x' ▸ dist_comm y' x' ▸ Hy')
+end
+
+/-- Emetric diameter of the convex hull of a set `s` equals the emetric diameter of `s. -/
+@[simp] lemma convex_hull_ediam (s : set E) :
+  emetric.diam (convex_hull ℝ s) = emetric.diam s :=
+begin
+  refine (emetric.diam_le $ λ x hx y hy, _).antisymm (emetric.diam_mono $ subset_convex_hull ℝ s),
+  rcases convex_hull_exists_dist_ge2 hx hy with ⟨x', hx', y', hy', H⟩,
+  rw edist_dist,
+  apply le_trans (ennreal.of_real_le_of_real H),
+  rw ← edist_dist,
+  exact emetric.edist_le_diam_of_mem hx' hy'
+end
+
+/-- Diameter of the convex hull of a set `s` equals the emetric diameter of `s. -/
+@[simp] lemma convex_hull_diam (s : set E) :
+  metric.diam (convex_hull ℝ s) = metric.diam s :=
+by simp only [metric.diam, convex_hull_ediam]
+
+/-- Convex hull of `s` is bounded if and only if `s` is bounded. -/
+@[simp] lemma bounded_convex_hull {s : set E} :
+  metric.bounded (convex_hull ℝ s) ↔ metric.bounded s :=
+by simp only [metric.bounded_iff_ediam_ne_top, convex_hull_ediam]
+
+@[priority 100]
+instance normed_space.path_connected : path_connected_space E :=
+topological_add_group.path_connected
+
+@[priority 100]
+instance normed_space.loc_path_connected : loc_path_connected_space E :=
+loc_path_connected_of_bases (λ x, metric.nhds_basis_ball)
+  (λ x r r_pos, (convex_ball x r).is_path_connected $ by simp [r_pos])
+
+lemma dist_add_dist_of_mem_segment {x y z : E} (h : y ∈ [x -[ℝ] z]) :
+  dist x y + dist y z = dist x z :=
+begin
+  simp only [dist_eq_norm, mem_segment_iff_same_ray] at *,
+  simpa only [sub_add_sub_cancel', norm_sub_rev] using h.norm_add.symm
+end
+
+/-- The set of vectors in the same ray as `x` is connected. -/
+lemma is_connected_set_of_same_ray (x : E) : is_connected {y | same_ray ℝ x y} :=
+begin
+  by_cases hx : x = 0, { simpa [hx] using is_connected_univ },
+  simp_rw ←exists_nonneg_left_iff_same_ray hx,
+  exact is_connected_Ici.image _ ((continuous_id.smul continuous_const).continuous_on)
+end
+
+/-- The set of nonzero vectors in the same ray as the nonzero vector `x` is connected. -/
+lemma is_connected_set_of_same_ray_and_ne_zero {x : E} (hx : x ≠ 0) :
+  is_connected {y | same_ray ℝ x y ∧ y ≠ 0} :=
+begin
+  simp_rw ←exists_pos_left_iff_same_ray_and_ne_zero hx,
+  exact is_connected_Ioi.image _ ((continuous_id.smul continuous_const).continuous_on)
+end
```

Comment:
Although I originally proved these in a normed context, they should work in a more general topological context (with `haveI : path_connected_space E := topological_add_group.path_connected,` added to the first proof so that `is_connected_univ` works), without actually needing norms.